eip/EIPS/eip-4337.md

```diff
@@ -391,78 +451,104 @@ The result `SHOULD` be set to true if and only if the request passed simulation
+**Parameters**: just like `eth_sendUserOperation`
+**Return Value**:
+* **preOpGas** gas used by validation (including preVerificationGas)
+* **prefund** amount the wallet need to pre-pay (above preOpGas and callGasLimit)
+* **deadline** if set, until what time this userOp is valid
+* **paymasterInfo** if set, current paymasterStake and paymasterUnstakeDelay
+* **aggregator** if set, the signature-aggregator used by the wallet.
 
-The entry point-based approach allows for a clean separation between verification and execution, and keeps wallets' logic simple. The alternative would be to require wallets to follow a template where they first self-call to verify and then self-call to execute (so that the execution is sandboxed and cannot cause the fee payment to revert); template-based approaches were rejected due to being harder to implement, as existing code compilation and verification tooling is not designed around template verification.
+**Error Codes:**: just like `eth_sendUserOperation`
 
-### Paymasters
+### * eth_getUserOperationTransactionByHash
 
-Paymasters facilitate transaction sponsorship, allowing third-party-designed mechanisms to pay for transactions. Many of these mechanisms _could_ be done by having the paymaster wrap a `UserOperation` with their own, but there are some important fundamental limitations to that approach:
+Return a transaction information based on a UserOperation hash (requestId) returned by `eth_sendUserOperation`
 
-* No possibility for "passive" paymasters (eg. that accept fees in some ERC-20 token at an exchange rate pulled from an on-chain DEX)
-* Paymasters run the risk of getting griefed, as users could send ops that appear to pay the paymaster but then change their behavior after a block
+An equivalent of `eth_getTransactionByHash`
 
-The paymaster scheme allows a contract to passively pay on users' behalf under arbitrary conditions. It even allows ERC-20 token paymasters to secure a guarantee that they would only need to pay if the user pays them: the paymaster contract can check that there is sufficient approved ERC-20 balance in the `validatePaymasterUserOp` method, and then extract it with `transferFrom` in the `postOp` call; if the op itself transfers out or de-approves too much of the ERC-20s, the inner `postOp` will fail and revert the execution and the outer `postOp` can extract payment (note that because of storage access restrictions the ERC-20 would need to be a wrapper defined within the paymaster itself).
+**Parameters**
+* **hash** a requestId value returned by `eth_sendUserOperation`
 
-### First-time wallet creation
+**Return value**:
```

Comment:
I suggest creating a wrapped object that has original UserOperation, EntryPoint transaction, `requestId` and other relevant information here. Otherwise, this call may end up being quite useless.

Reply:
Look at other rpc APIs. It's but a high level language, and they don't introduce objects just because. A UserOp is an object, like tx, but other structures are not added just for the sake of the api